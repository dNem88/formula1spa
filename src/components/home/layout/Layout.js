import React, {Fragment} from 'react'
import styles from './layout.module.css'
import News from '../news/News'

function Layout(props) {
    return (
        <Fragment>
            <section className={styles['trending']}>

            </section>
            <section className={styles['news']}>
                < News  content = {
                    'Dani Nemski is supercool'
                }
                imageUrl = {
                    'https://www.formula1.com/content/dam/fom-website/manual/Misc/2022manual/WinterFebruary/HaasVF22Launch/MAZ%20Top.jpg.transform/6col/image.jpg'
                }
                />
                < News content = {
                    'Dani Nemski is supercool'
                }
                imageUrl = {
                    'https://www.formula1.com/content/dam/fom-website/manual/Misc/2022manual/WinterFebruary/HaasVF22Launch/MAZ%20Top.jpg.transform/6col/image.jpg'
                }
                />
                < News content = {
                    'Dani Nemski is supercool'
                }
                imageUrl = {
                    'https://www.formula1.com/content/dam/fom-website/manual/Misc/2022manual/WinterFebruary/HaasVF22Launch/MAZ%20Top.jpg.transform/6col/image.jpg'
                }
                />
                < News content = {
                    'Dani Nemski is supercool'
                }
                imageUrl = {
                    'https://www.formula1.com/content/dam/fom-website/manual/Misc/2022manual/WinterFebruary/HaasVF22Launch/MAZ%20Top.jpg.transform/6col/image.jpg'
                }
                />
            </section>
            <fieldset className={styles['editor-pick']}>
                <legend className={styles['editor-legend']}>EDITOR'S PICK</legend>
            </fieldset>
            <h2 className={styles['h-news']}>More news</h2>
            <section className={styles['more-news']}>
                < News hoverChange={true} content = {
                    'Bottas and Hamilton greatest strength after their 5 years in Mercedes'
                }
                imageUrl = {
                    'https://www.formula1.com/content/dam/fom-website/manual/Misc/2022manual/WinterJanuary/GettyImages-1234043233.jpg.transform/6col/image.jpg'
                }
                />
                < News hoverChange = {
                    true
                }
                content = {
                    'Bottas and Hamilton greatest strength after their 5 years in Mercedes'
                }
                imageUrl = {
                    'https://www.formula1.com/content/dam/fom-website/manual/Misc/2022manual/WinterJanuary/GettyImages-1234043233.jpg.transform/6col/image.jpg'
                }
                />
                < News hoverChange = {
                    true
                }
                content = {
                    'Bottas and Hamilton greatest strength after their 5 years in Mercedes'
                }
                imageUrl = {
                    'https://www.formula1.com/content/dam/fom-website/manual/Misc/2022manual/WinterJanuary/GettyImages-1234043233.jpg.transform/6col/image.jpg'
                }
                />
                < News hoverChange={true} content = {
                    'Bottas and Hamilton greatest strength after their 5 years in Mercedes'
                }
                imageUrl = {
                    'https://www.formula1.com/content/dam/fom-website/manual/Misc/2022manual/WinterJanuary/GettyImages-1234043233.jpg.transform/6col/image.jpg'
                }
                />
               
            </section>
            <section className={styles['schedule']}>

            </section>
            <section className={styles['standings']}>

            </section>
        </Fragment>
    )
}

export default Layout