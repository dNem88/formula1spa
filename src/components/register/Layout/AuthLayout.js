import React from 'react'
import {Outlet} from 'react-router-dom'
import SelectForm from '../Select/SelectForm'
import styles from './authLayout.module.css'
function AuthLayout() {
    return (
        <div className={styles.container}>
            <SelectForm location={'auth'}/>
            <div className={styles['form-container']}>
                <Outlet/>
            </div>
        </div>
    )
}

export default AuthLayout