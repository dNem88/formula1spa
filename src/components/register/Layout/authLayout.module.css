.container{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
}
.form-container{
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: #e4e1dd;
}