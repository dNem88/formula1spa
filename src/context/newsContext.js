import React from 'react';

const newsContext = React.createContext({articles: []})

export default newsContext;