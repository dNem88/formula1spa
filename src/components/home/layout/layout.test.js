import React from 'react'
import renderer from 'react-test-renderer'
import newsContext from '../../../context/newsContext'
import {MemoryRouter} from 'react-router-dom'
import Layout from './Layout'
let contextMock = {articles: [
    {_id: 1, title: 'Dani nemski', urltoImage: 'test-url'},
    {_id: 2, title: 'daniel2', urltoImage: 'test-url'},
    {_id: 3, title: 'Dani nemski', urltoImage: 'test-url'},
    {_id: 4, title: 'Dani nemski', urltoImage: 'test-url'},
    {_id: 5, title: 'Dani nemski', urltoImage: 'test-url'},
    {_id: 6, title: 'Dani nemski', urltoImage: 'test-url'},
    {_id: 7, title: 'Dani nemski', urltoImage: 'test-url'},
    {_id: 8, title: 'Dani nemski', urltoImage: 'test-url'},
    {_id: 9, title: 'Dani nemski', urltoImage: 'test-url'},
    {_id: 10, title: 'Dani nemski', urltoImage: 'test-url'},
    {_id: 11, title: 'Dani nemski', urltoImage: 'test-url'},
    {
        _id: 1,
        title: 'Dani nemski',
        urltoImage: 'test-url'
    }
]}
describe('Testing Layout Component', () => {
    it('Should render properly', () => {
        let layout = renderer.create(
            <MemoryRouter>
                <newsContext.Provider value={contextMock}>
                    <Layout/>
                </newsContext.Provider>
            </MemoryRouter>
        ).toJSON();
        expect(layout).toMatchSnapshot();
        expect(layout[0].type).toBe('section')
        expect(layout[0].props.className).toBe('trending')
        expect(layout[0].children.length).toBe(1)
        expect(layout[1].type).toBe('section')
        expect(layout[1].props.className).toBe('news')
        expect(layout[1].children.length).toBe(4)
        expect(layout[2].type).toBe('fieldset')
        expect(layout[2].props.className).toBe('fieldset-full')
        expect(layout[3].type).toBe('h2')
        expect(layout[3].props.className).toBe('h-news')
        expect(layout[4].type).toBe('section')
        expect(layout[4].props.className).toBe('more-news')
        expect(layout[4].children.length).toBe(4)
        expect(layout[5].type).toBe('div')
        expect(layout[6].type).toBe('section')
        expect(layout[6].props.className).toBe('standings')
    })
})