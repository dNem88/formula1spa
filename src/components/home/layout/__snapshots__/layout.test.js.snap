// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Layout Component Should render properly 1`] = `
Array [
  <section
    className="trending"
  >
    <div
      className="wrapper"
      id={1}
      onClick={[Function]}
    >
      <p
        className="trending-p"
      >
        TRENDING
      </p>
      <h1
        className="trending-heading"
      >
        Dani nemski
      </h1>
      <div
        className="image-container"
      >
        <img
          alt="someAlt"
          className="trending-image"
        />
      </div>
    </div>
  </section>,
  <section
    className="news"
  >
    <div
      className="main-container"
      id={2}
      onClick={[Function]}
    >
      <div
        className="image-container"
      >
        <img
          alt="someAlt"
          id="news-image"
        />
      </div>
      <div
        className="container"
      >
        <p
          className="news-par"
        >
          NEWS
        </p>
        <p
          className="content-par"
        >
          daniel2
        </p>
      </div>
    </div>
    <div
      className="main-container"
      id={3}
      onClick={[Function]}
    >
      <div
        className="image-container"
      >
        <img
          alt="someAlt"
          id="news-image"
        />
      </div>
      <div
        className="container"
      >
        <p
          className="news-par"
        >
          NEWS
        </p>
        <p
          className="content-par"
        >
          Dani nemski
        </p>
      </div>
    </div>
    <div
      className="main-container"
      id={4}
      onClick={[Function]}
    >
      <div
        className="image-container"
      >
        <img
          alt="someAlt"
          id="news-image"
        />
      </div>
      <div
        className="container"
      >
        <p
          className="news-par"
        >
          NEWS
        </p>
        <p
          className="content-par"
        >
          Dani nemski
        </p>
      </div>
    </div>
    <div
      className="main-container"
      id={5}
      onClick={[Function]}
    >
      <div
        className="image-container"
      >
        <img
          alt="someAlt"
          id="news-image"
        />
      </div>
      <div
        className="container"
      >
        <p
          className="news-par"
        >
          NEWS
        </p>
        <p
          className="content-par"
        >
          Dani nemski
        </p>
      </div>
    </div>
  </section>,
  <fieldset
    className="editor-pick"
  >
    <legend
      className="editor-legend"
    >
      EDITOR'S PICK
    </legend>
  </fieldset>,
  <h2
    className="h-news"
  >
    More news
  </h2>,
  <section
    className="more-news"
  >
    <div
      className="main-container"
      id={6}
      onClick={[Function]}
    >
      <div
        className="image-container"
      >
        <img
          alt="someAlt"
          id="news-image"
        />
      </div>
      <div
        className="dark-container"
      >
        <p
          className="news-par"
        >
          NEWS
        </p>
        <p
          className="content-par"
        >
          Dani nemski
        </p>
      </div>
    </div>
    <div
      className="main-container"
      id={7}
      onClick={[Function]}
    >
      <div
        className="image-container"
      >
        <img
          alt="someAlt"
          id="news-image"
        />
      </div>
      <div
        className="dark-container"
      >
        <p
          className="news-par"
        >
          NEWS
        </p>
        <p
          className="content-par"
        >
          Dani nemski
        </p>
      </div>
    </div>
    <div
      className="main-container"
      id={8}
      onClick={[Function]}
    >
      <div
        className="image-container"
      >
        <img
          alt="someAlt"
          id="news-image"
        />
      </div>
      <div
        className="dark-container"
      >
        <p
          className="news-par"
        >
          NEWS
        </p>
        <p
          className="content-par"
        >
          Dani nemski
        </p>
      </div>
    </div>
    <div
      className="main-container"
      id={9}
      onClick={[Function]}
    >
      <div
        className="image-container"
      >
        <img
          alt="someAlt"
          id="news-image"
        />
      </div>
      <div
        className="dark-container"
      >
        <p
          className="news-par"
        >
          NEWS
        </p>
        <p
          className="content-par"
        >
          Dani nemski
        </p>
      </div>
    </div>
  </section>,
  <section
    className="schedule"
  />,
  <section
    className="standings"
  />,
]
`;
